public/app/plugins/datasource/grafana-azure-monitor-datasource/components/QueryEditor/useDefaultQuery.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useMemo } from 'react';
+import { AzureMonitorQuery, AzureQueryType } from '../../types';
+
+const DEFAULT_QUERY_TYPE = AzureQueryType.AzureMonitor;
+
+const createQueryWithDefaults = (query: AzureMonitorQuery) => {
+  // A quick and easy way to set just the default query type. If we want to set any other defaults,
+  // we might want to look into something more robust
+  if (!query.queryType) {
+    return {
+      ...query,
+      queryType: query.queryType ?? DEFAULT_QUERY_TYPE,
+    };
+  }
+
+  return query;
+};
+
+/**
+ * Returns queries with some defaults, and calls onChange function to notify if it changes
+ */
+const useDefaultQuery = (query: AzureMonitorQuery, onChangeQuery: (newQuery: AzureMonitorQuery) => void) => {
+  const queryWithDefaults = useMemo(() => createQueryWithDefaults(query), [query]);
+
+  useEffect(() => {
+    if (queryWithDefaults !== query) {
```

Comment:
should we use lodash or something to do a better object comparison?

Reply:
Yeah... that was kind of what I was thinking with my comment in `createQueryWithDefaults`. I'll explain my thinking:

 - I wanted the query to have the defaults on the initial render, so it could flow down to the components and no potential flash of it changing. Especially important for the queryType as it changes the whole UI basically.
 - If no defaults need to be set, the identity of `query` should remain the same, so effects and other things arent triggered on each render
   - This is the important part - returning the same object if nothing changed.
 - I _think_ that when the defaults are set, we should call `onChangeQuery` to tell the panel about the new values. I think this is debatable - strictly speaking i guess it doesn't need to happen?

I thought about using `_.defaultsDeep`, but  defaultsDeep will always change the identity of the query, regardless on the changes made to it. So we would need to do a deep comparison, something like...

```js
const queryWithDefaults = _.defaultsDeep({...query}, queryDefaults);
if (deepEqual(query, queryWithDefaults)) {
  return query;
} else {
  return queryWithDefaults
}
```

on every render (or with `useMemo` only when `query` changes). I thought all of this was a smidge too much right now when the only default is the query type.

Definitely open to your thoughts on this - my mind isn't made in stone for the approach here.

Reply:
Ohhh ok I think I'm following now, the useMemo is kind of doing the heavy lifting for us then? I think this should be fine!